hazelcast-sql/src/test/java/com/hazelcast/sql/SqlBasicTest.java

```diff
@@ -0,0 +1,961 @@
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.UseParametersRunnerFactory;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * Test that covers basic column read operations through SQL.
+ */
+@RunWith(Parameterized.class)
+@UseParametersRunnerFactory(HazelcastParallelParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:RedundantModifier")
+public class SqlBasicTest extends SqlTestSupport {
+
+    private static final int IDS_FACTORY_ID = 1;
+    private static final int IDS_KEY_CLASS_ID = 2;
+    private static final int IDS_VALUE_CLASS_ID = 3;
+    private static final int PORTABLE_FACTORY_ID = 1;
+    private static final int PORTABLE_KEY_CLASS_ID = 2;
+    private static final int PORTABLE_VALUE_CLASS_ID = 3;
+    private static final int PORTABLE_NESTED_CLASS_ID = 4;
+
+    private static final String MAP_OBJECT = "map_object";
+    private static final String MAP_BINARY = "map_binary";
+
+    private static final int[] PAGE_SIZES = { 1, 16, 256, 4096 };
+    private static final int[] DATA_SET_SIZES = { 1, 256, 4096 };
+    private static final TestHazelcastInstanceFactory FACTORY = new TestHazelcastInstanceFactory(2);
+
+    private static HazelcastInstance instance;
+
+    @Parameter
+    public int cursorBufferSize;
+
+    @Parameter(1)
+    public int dataSetSize;
+
+    @Parameter(2)
+    public SerializationMode serializationMode;
+
+    @Parameter(3)
+    public InMemoryFormat inMemoryFormat;
+
+    @Parameters(name = "cursorBufferSize:{0}, dataSetSize:{1}, serializationMode:{2}, inMemoryFormat:{3}")
+    public static Collection<Object[]> parameters() {
+        List<Object[]> res = new ArrayList<>();
+
+        for (int pageSize : PAGE_SIZES) {
+            for (int dataSetSize : DATA_SET_SIZES) {
+                for (SerializationMode serializationMode : SerializationMode.values()) {
+                    for (InMemoryFormat format : new InMemoryFormat[] { InMemoryFormat.OBJECT, InMemoryFormat.BINARY }) {
```

Comment:
Shall we create a similar run for HD?

Reply:
HD is not supported in the scope of this PR